File: leetcode_921/MinimumAddToMakeParenthesesValid.java
package leetcode_921;

class MinimumAddToMakeParenthesesValid {
    public int minAddToMakeValid(String s) {
        int leftCount = 0;
        int rightCount = 0;
        for (char c : s.toCharArray()) {
            if (c == '(') {
                leftCount++;
            } else if (leftCount > 0) {
                leftCount--;
            } else {
                rightCount++;
            }
        }
        return leftCount + rightCount;
    }

    public static void main(String[] args) {
        MinimumAddToMakeParenthesesValid obj = new MinimumAddToMakeParenthesesValid();
        System.out.println(obj.minAddToMakeValid("())"));
        System.out.println(obj.minAddToMakeValid("((("));
    }
}

File: leetcode_921/MinimumAddToMakeParenthesesValid.py
class MinimumAddToMakeParenthesesValid:
    def min_add_to_make_valid(self, s: str) -> int:
        leftCount, rightCount = 0, 0
        for c in s:
            if c == '(':
                leftCount += 1
            elif leftCount > 0:
                leftCount -= 1
            else:
                rightCount += 1
        return leftCount + rightCount


def main():
    main_instance = MinimumAddToMakeParenthesesValid()
    print(main_instance.min_add_to_make_valid("())"))
    print(main_instance.min_add_to_make_valid("((("))


if __name__ == "__main__":
    main()